src/socket/tcp.rs

```diff
@@ -1558,7 +1558,8 @@ impl<'a> TcpSocket<'a> {
                 // Extract as much data as the remote side can receive in this packet
                 // from the transmit buffer.
                 let offset = self.remote_last_seq - self.local_seq_no;
-                let size = cmp::min(self.remote_win_len, self.remote_mss);
+                let size = cmp::min(cmp::min(self.remote_win_len, self.remote_mss),
+                     caps.max_transmission_unit);
```

Comment:
I'm not convinced this is correct. MTU is IP packet size (IP header + TCP header + payload). `size` here is only payload size.

You have to subtract the IP and TCP header sizes, as in here : https://github.com/smoltcp-rs/smoltcp/blob/1268fb3d8ddeb1814a6b3a49f1647d1bbf3debd8/src/socket/tcp.rs#L1624

Reply:
Ah, so much for a quick fix on my part. Haha. 

This is fine: 
```rust
let size = cmp::min(cmp::min(self.remote_win_len, self.remote_mss),
                     caps.max_transmission_unit - ip_repr.buffer_len() - repr.mss_header_len());
                repr.payload = self.tx_buffer.get_allocated(offset, size);
```

Reply:
Sure!